spec/lib/tasks/world_taxon_update_helper_spec.rb

```diff
@@ -0,0 +1,268 @@
+    end
+
+    it "skips taxons with internal names that include '(GENERIC)'" do
+      allow_item_double(linked_item_generic)
+      expect(Taxonomy::BuildTaxon).not_to receive(:call)
+      expect(Taxonomy::UpdateTaxon).not_to receive(:call)
+      expect(described_class.new).not_to receive(:create_new_taxon_title).with(linked_item_generic.internal_name)
+
+      expect { described_class.new.add_country_names("tmp/rake_log") }.to output(/Skipping Doing a thing \(GENERIC\)/).to_stdout
+    end
+  end
+
+  describe "remove_country_name_from_title" do
+    # Set up the already processed taxons we will need to revert (e.g. titles including the COUNTRY_NAME)
+    let(:retitled_child_item_coming_to) do
+      Taxonomy::LinkedContentItem.new(
+        base_path: "/root/taxon_1",
+        content_id: "id3",
+        title: "Coming to the UK from Porpoise Spit",
+        internal_name: "Coming to the UK (Porpoise Spit)",
+      )
+    end
+    let(:retitled_child_item_trade_invest) do
+      Taxonomy::LinkedContentItem.new(
+        base_path: "/root/taxon_2",
+        content_id: "id4",
+        title: "Trade and invest: Porpoise Spit",
+        internal_name: "Trade and invest (Porpoise Spit)",
+      )
+    end
+    let(:retitled_child_item_birth_death) do
+      Taxonomy::LinkedContentItem.new(
+        base_path: "/root/taxon_3",
+        content_id: "id5",
+        title: "Birth, death and marriage abroad in Porpoise Spit",
+        internal_name: "Birth, death and marriage abroad (Porpoise Spit)",
+      )
+    end
+    let(:retitled_multi_level_linked_items) do
+      root_item << parent_item_uk_help
+      parent_item_uk_help << retitled_child_item_coming_to << retitled_child_item_trade_invest << retitled_child_item_birth_death
+
+      root_item
+    end
+
+    let(:retitled_child_taxon_coming_to) do
+      FactoryBot.build(:taxon, content_id: "id3", title: "Coming to the UK from Porpoise Spit")
+    end
+    let(:retitled_child_taxon_trade_invest) do
+      FactoryBot.build(:taxon, content_id: "id4", title: "Trade and invest: Porpoise Spit")
+    end
+    let(:retitled_child_taxon_birth_death) do
+      FactoryBot.build(:taxon, content_id: "id5", title: "Birth, death and marriage abroad in Porpoise Spit")
+    end
+
+    it "logs an error to stderr if the taxon is invalid" do
+      allow_item_double(retitled_multi_level_linked_items)
+      allow(Taxonomy::BuildTaxon).to receive(:call).and_raise(Taxonomy::UpdateTaxon::InvalidTaxonError.new("Invalid taxon"))
+
+      expect { described_class.new.remove_country_names("tmp/rake_log") }.to output(/An error occurred while processing taxon/).to_stderr
+    end
+
+    it "logs an error to stderr if BulkPublishTaxon fails" do
+      allow_item_double(retitled_multi_level_linked_items)
+      allow(Taxonomy::BuildTaxon).to receive(:call).and_return(retitled_child_taxon_coming_to)
```

Comment:
Similar to the test for add, this test doesn't really care what the taxon returned by BuildTaxon is, so we could change this line (which then allows the 3 variables at the top of the describe block to be moved down into the remaining test that uses them).
```suggestion
      allow(Taxonomy::BuildTaxon).to receive(:call).and_return(FactoryBot.build(:taxon))
```